nixos/modules/services/mail/postsrsd.nix

```diff
@@ -107,29 +90,42 @@ in
       postsrsd.gid = config.ids.gids.postsrsd;
     };
 
+    systemd.services.postsrsd-generate-secrets = {
+      path = [ pkgs.coreutils ];
+      script = ''
+        if [ -e "${cfg.secretsFile}" ]; then
+          echo "Secrets file exists. Nothing to do!"
+        else
+          echo "WARNING: secrets file not found, autogenerating!"
+          DIR="$(dirname "${cfg.secretsFile}")"
+          install -m 750 -o ${cfg.user} -g ${cfg.group} -d "$DIR"
+          install -m 600 -o ${cfg.user} -g ${cfg.group} <(dd if=/dev/random bs=18 count=1 | base64) "${cfg.secretsFile}"
+        fi
+      '';
+      serviceConfig = {
+        Type = "oneshot";
+      };
+    };
+
     systemd.services.postsrsd = {
       description = "PostSRSd SRS rewriting server";
-      after = [ "network.target" ];
+      after = [
+        "network.target"
+        "postsrsd-generate-secrets.service"
+      ];
       before = [ "postfix.service" ];
       wantedBy = [ "multi-user.target" ];
-
-      path = [ pkgs.coreutils ];
+      requires = [ "postsrsd-generate-secrets.service" ];
+      confinement.enable = true;
 
       serviceConfig = {
-        ExecStart = ''${pkgs.postsrsd}/sbin/postsrsd "-s${cfg.secretsFile}" "-d${cfg.domain}" -a${cfg.separator} -f${toString cfg.forwardPort} -r${toString cfg.reversePort} -t${toString cfg.timeout} "-X${lib.concatStringsSep "," cfg.excludeDomains}"'';
+        ExecStart = "${lib.getExe pkgs.postsrsd} -C ${configFile}";
         User = cfg.user;
         Group = cfg.group;
         PermissionsStartOnly = true;
+        RuntimeDirectory = runtimeDirectoryName;
```

Comment:
```suggestion
        RuntimeDirectory = = "postsrsd";;
```

Reply:
Marking resolved as per the conversation over here: https://github.com/NixOS/nixpkgs/pull/397316/files#r2040689526